pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/phimanifest/PhiManifestService.java

```diff
@@ -0,0 +1,131 @@
+        }
+        if (!isParticipantConsented(ddpParticipantId, ddpInstanceDto)) {
+            maybeErrorMessage = String.format("Participant %s has not consented to receive shared learning ", ddpParticipantId);
+        }
+        if (StringUtils.isNotBlank(maybeErrorMessage)) {
+            log.warn(maybeErrorMessage);
+            throw new DSMBadRequestException(maybeErrorMessage);
+        }
+        return generateReport(ddpParticipantId, sequencingOrderNumber, ddpInstanceDto);
+    }
+
+    private Object generateReport(String ddpParticipantId, String sequencingOrderNumber, DDPInstanceDto ddpInstanceDto) {
+        return null;
+    }
+
+    public boolean isParticipantConsented(@NonNull String ddpParticipantId, @NonNull DDPInstanceDto ddpInstanceDto) {
+        ElasticSearchParticipantDto participant = ElasticSearchUtil.getParticipantESDataByParticipantId(
+                ddpInstanceDto.getEsParticipantIndex(), ddpParticipantId);
+        return hasParticipantConsentedToSharedLearning(participant) && hasParticipantConsentedToTumor(participant,
+                ddpInstanceDto.getStudyGuid());
+    }
+
+    private boolean hasParticipantConsentedToTumor(ElasticSearchParticipantDto participant, String studyGuid) {
+        if (DBConstants.LMS_STUDY_GUID.equals(studyGuid)) {
+            return checkAnswerToActivity(participant, "CONSENT_ADDENDUM", "SOMATIC_CONSENT_ADDENDUM_TUMOR", true);
+        } else if (DBConstants.OSTEO_STUDY_GUID.equals(studyGuid)) {
+            return checkAnswerToActivity(participant, "CONSENT_ADDENDUM", "SOMATIC_CONSENT_TUMOR", true);
+        }
+        return false;
+    }
+
+    private boolean hasParticipantConsentedToSharedLearning(ElasticSearchParticipantDto participant) {
+        String dateOfBirth = participant.getDsm().get().getDateOfBirth();
+        int age = calculateAge(dateOfBirth);
+        if (age >= 7) {
+            return checkAnswerToActivity(participant, "CONSENT_ADDENDUM_PEDIATRIC", "SOMATIC_CONSENT_TUMOR_PEDIATRIC",
+                    true) && checkAnswerToActivity(participant, "CONSENT_ADDENDUM_PEDIATRIC",
+                    "SOMATIC_ASSENT_ADDENDUM", true);
+
+        } else if (age < 7) {
+            return checkAnswerToActivity(participant, "CONSENT_ADDENDUM_PEDIATRIC", "SOMATIC_CONSENT_TUMOR_PEDIATRIC",
+                    true);
+        }
+        return false;
+    }
+
+    /** checks if the answer provided by a participant to a survey question matches the criteria
+     * @param participant participant data from ES.
+     * */
+    private boolean checkAnswerToActivity(ElasticSearchParticipantDto participant, String activityName, String question, Object answer) {
+        Optional<Activities> maybeActivity = participant.getActivities().stream().filter(activities -> activities.getActivityCode()
+                .equals(activityName)).findAny();
+        if (maybeActivity.isPresent()) {
+            Activities activities = maybeActivity.get();
```

Comment:
Consider moving the guts of this method into `activities.hasAnswerToQuestion(String questionId, Object answer)`.  Then we can add tests to `Activities` and get re-use of the activity object.